src/core/utility/file/filePath/FilePath.cpp

```diff
@@ -170,27 +156,27 @@ FilePath& FilePath::makeCanonical() {
     } else {
       canonicalPath /= *it;
 
-      if(boost::filesystem::is_symlink(boost::filesystem::symlink_status(canonicalPath))) {
-        boost::filesystem::path symlink = boost::filesystem::read_symlink(canonicalPath);
+      if(fs::is_symlink(fs::symlink_status(canonicalPath))) {
+        fs::path symlink = fs::read_symlink(canonicalPath);
         if(!symlink.empty()) {
           // on Windows the read_symlink function discards the drive letter (this is a
           // boost bug). Therefore we need to make the path absolute again. We also have
           // to discard the trailing \0 characters so that we can continue appending to
           // the path.
-          canonicalPath = utility::substrBeforeFirst(boost::filesystem::absolute(symlink).string(), '\0');
+          canonicalPath = utility::substrBeforeFirst(fs::absolute(symlink).string(), '\0');
         }
       }
     }
   }
 #else
   try {
-    canonicalPath = boost::filesystem::canonical(getPath());
-  } catch(boost::filesystem::filesystem_error& e) {
+    canonicalPath = fs::canonical(getPath());
+  } catch(fs::filesystem_error& e) {
     LOG_ERROR(e.what());
     return *this;
   }
```

Comment:
Remove try, catch and use std::errorCode;
```cpp
std::errorCode errorCode;
canonicalPath = fs::canonical(getPath(), errorCode);
```

Reply:
I agree.